notebook/metrics.py

```diff
@@ -0,0 +1,39 @@
+"""
+Prometheus metrics exported by Jupyter Notebook Server
+
+Read https://prometheus.io/docs/practices/naming/ for naming
+conventions for metrics & labels. We generally prefer naming them
+`<noun>_<verb>_<type_suffix>`. So a histogram that's tracking
+the duration (in seconds) of servers spawning would be called
+SERVER_SPAWN_DURATION_SECONDS.
```

Comment:
copy/paste. REQUEST_DURATION_SECONDS

Reply:
As an FYI, this particular example breaks the naming rule in the docstring.

Reply:
Perhaps better to remove the preference sentence with noun/verb/type.

Consider renaming to `NOTEBOOK_REQUEST_DURATION_SECONDS` based on Prometheus docs.

Reply:
Actually, it's not clear to me what a 'request duration' is - is that the time from the request being sent to it being received? The time from receiving the first byte to receiving the last? The time from receiving the request to sending the response?

If this is a standard term in web metrics, it doesn't matter that it's not familiar to me. But if it's a term we're creating, maybe we can create something less ambiguous.

Reply:
Ping @yuvipanda

Reply:
Heya!

I removed the naming convention recommendation, and just directly linked only to the page instead. This should hopefully reduce confusion.

I've also renamed this metric to http_request_duration_seconds. I think that is pretty standard for what we are doing here, which is indiscriminately recording metric info for *all* http requests. Operators usually use [job and instance](https://prometheus.io/docs/concepts/jobs_instances/) labels automatically added by prometheus to differentiate various applications & instances of applications. So I think in this case, it's ok to not use a prefix.